app/components/tutorial_highlight/item.tsx

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useMemo} from 'react';
+import {StyleSheet} from 'react-native';
+import Svg, {ClipPath, Defs, G, Path, Rect} from 'react-native-svg';
+import tinyColor from 'tinycolor2';
+
+import {useTheme} from '@context/theme';
+
+type Props = {
+    borderRadius?: number;
+    height: number;
+    itemBounds: TutorialItemBounds;
+    onDismiss: () => void;
+    width: number;
+}
+
+const svgM = (x: number, y: number) => `M ${x} ${y}`;
+const svgL = (x: number, y: number) => `L ${x} ${y}`;
+const svgArc = (toX: number, toY: number, radius: number) => `A ${radius},${radius} 0 0 0 ${toX},${toY}`;
+const z = 'z';
+const constructRectangularPath = (
+    parentBounds: TutorialItemBounds,
+    itemBounds: TutorialItemBounds,
+    borderRadius: number,
+): string => {
+    const {startX, startY, endX, endY} = itemBounds;
+    return [
+        svgM(parentBounds.startX, parentBounds.startY),
+        svgL(parentBounds.startX, parentBounds.endY),
+        svgL(parentBounds.endX, parentBounds.endY),
+        svgL(parentBounds.endX, parentBounds.startY),
+        z,
+        svgM(startX, startY + borderRadius),
+        svgL(startX, endY - borderRadius),
+        svgArc(startX + borderRadius, endY, borderRadius),
+        svgL(endX - borderRadius, endY),
+        svgArc(endX, endY - borderRadius, borderRadius),
+        svgL(endX, startY + borderRadius),
+        svgArc(endX - borderRadius, startY, borderRadius),
+        svgL(startX + borderRadius, startY),
+        svgArc(startX, startY + borderRadius, borderRadius),
+    ].join(' ');
+};
```

Comment:
These seem useful. Are we planning to use them somewhere else? (and therefore move them probably to utils)